pkg/splunk/enterprise/util.go

```diff
@@ -726,3 +789,24 @@ func setupAppInitContainers(client splcommon.ControllerClient, cr splcommon.Meta
 		}
 	}
 }
+
+// ConvertS3Response converts S3 Response to a mock client response
+func ConvertS3Response(s3Response splclient.S3Response) (spltest.MockAWSS3Client, error) {
+	scopedLog := log.WithName("ConvertS3Response")
+
+	var mockResponse spltest.MockAWSS3Client
+
+	tmp, err := json.Marshal(s3Response)
+	if err != nil {
+		scopedLog.Error(err, "Unable to marshal s3 response")
+		return mockResponse, err
+	}
+
+	err = json.Unmarshal(tmp, &mockResponse)
+	if err != nil {
+		scopedLog.Error(err, "Unable to unmarshal s3 response")
+		return mockResponse, err
+	}
+
+	return mockResponse, err
+}
```

Comment:
Can we move it to util_test.go, as it is test code?

Reply:
On a related note, i think we also need to introduce "normalized" responses for the actual case, just to be independent of multiple clients. Maybe sometime in a future sprint.